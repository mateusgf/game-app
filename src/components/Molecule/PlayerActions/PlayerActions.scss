.player-actions {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-around;
  align-items: center;
  align-content: center;

  &--action-wrp {
    &:hover {
      cursor: pointer;
      transform: scale(1.1);
      z-index: 2;
    }
  }

  &--action-label {
    display: none;
  }
}
