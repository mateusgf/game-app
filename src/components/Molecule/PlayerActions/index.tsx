import "./PlayerActions.scss";
import rockImage from "../../../assets/images/rock.jpg";
import paperImage from "../../../assets/images/paper.jpg";
import scissorsImage from "../../../assets/images/scissors.jpg";

interface IPlayerActionsProps {
  isHostPlayer: boolean;
  playAction: Function;
  isAvailableToPlayAction: boolean;
}

const PlayerActions = ({
  playAction,
  isHostPlayer,
  isAvailableToPlayAction,
}: IPlayerActionsProps) => {
  return isAvailableToPlayAction ? (
    <div className="player-actions" data-testid="player-actions">
      <div
        className="player-actions--action-wrp"
        data-testid="action-rock"
        onClick={() => playAction("rock", isHostPlayer)}
      >
        <img src={rockImage} alt="rock" />
        <span className="player-actions--action-label">rock</span>
      </div>

      <div
        className="player-actions--action-wrp"
        data-testid="action-paper"
        onClick={() => playAction("paper", isHostPlayer)}
      >
        <img src={paperImage} alt="paper" />
        <span className="player-actions--action-label">paper</span>
      </div>

      <div
        className="player-actions--action-wrp"
        data-testid="action-scissors"
        onClick={() => playAction("scissors", isHostPlayer)}
      >
        <img src={scissorsImage} alt="scissors" />
        <span className="player-actions--action-label">scissors</span>
      </div>
    </div>
  ) : (
    <h1 data-testid="waiting-opponent-label">Waiting for opponent to play</h1>
  );
};

export default PlayerActions;
