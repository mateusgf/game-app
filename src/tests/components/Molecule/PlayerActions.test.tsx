import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import PlayerActions from "../../../components/Molecule/PlayerActions";

describe("PlayerActions molecule component", () => {
  test("match snapshot", () => {

    const mockedProps = {
      playAction: jest.fn(),
      isHostPlayer: true,
      isAvailableToPlayAction: true,
    }

    const { container } = render(<PlayerActions {...mockedProps} />);
    expect(container).toMatchSnapshot();
  });

  test("rock action", async () => {

    const mockPlayAction = jest.fn();

    const mockedProps = {
      playAction: mockPlayAction,
      isHostPlayer: true,
      isAvailableToPlayAction: true,
    }

    render(<PlayerActions {...mockedProps} />);

    fireEvent(
      screen.getByTestId("action-rock"),
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      }),
    );

    const isHostPlayer = true;
    await waitFor(() => expect(mockPlayAction).toBeCalledWith("rock", isHostPlayer));
  });

  test("paper action", async () => {

    const mockPlayAction = jest.fn();

    const mockedProps = {
      playAction: mockPlayAction,
      isHostPlayer: true,
      isAvailableToPlayAction: true,
    }

    render(<PlayerActions {...mockedProps} />);

    fireEvent(
      screen.getByTestId("action-paper"),
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      }),
    );

    const isHostPlayer = true;
    await waitFor(() => expect(mockPlayAction).toBeCalledWith("paper", isHostPlayer));
  });

  test("scissors action", async () => {

    const mockPlayAction = jest.fn();

    const mockedProps = {
      playAction: mockPlayAction,
      isHostPlayer: true,
      isAvailableToPlayAction: true,
    }

    render(<PlayerActions {...mockedProps} />);

    fireEvent(
      screen.getByTestId("action-scissors"),
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      }),
    );

    const isHostPlayer = true;
    await waitFor(() => expect(mockPlayAction).toBeCalledWith("scissors", isHostPlayer));
  });

});
