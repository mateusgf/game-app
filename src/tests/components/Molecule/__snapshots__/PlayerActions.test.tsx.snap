// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerActions molecule component match snapshot 1`] = `
<div>
  <div
    class="player-actions"
    data-testid="player-actions"
  >
    <div
      class="player-actions--action-wrp"
      data-testid="action-rock"
    >
      <img
        alt="rock"
        src="rock.jpg"
      />
      <span
        class="player-actions--action-label"
      >
        rock
      </span>
    </div>
    <div
      class="player-actions--action-wrp"
      data-testid="action-paper"
    >
      <img
        alt="paper"
        src="paper.jpg"
      />
      <span
        class="player-actions--action-label"
      >
        paper
      </span>
    </div>
    <div
      class="player-actions--action-wrp"
      data-testid="action-scissors"
    >
      <img
        alt="scissors"
        src="scissors.jpg"
      />
      <span
        class="player-actions--action-label"
      >
        scissors
      </span>
    </div>
  </div>
</div>
`;
